lib/lookup.js

```diff
@@ -0,0 +1,80 @@
+/* eslint max-params: 0*/
+const semver = require('semver');
+
+/**
+ * Lookup Object for parsing build heads
+ * @public
+ * @constructor
+ * @param {Object} pkg - package object
+ * @param {Object[]} heads - build heads for a given package in the fetched env
+ */
+function Lookup(pkg, heads) {
```

Comment:
This is clean, well encapsulated logic. Semantically it's a bit confusing because `Lookup` is a verb. Tried to brainstorm a more obvious name and I came up with `BuildQuery`. I'm fine leaving it as-is (`Lookup`), but wanted to get your feel on that name.